consensus/src/persistent_liveness_storage.rs

```diff
@@ -127,9 +134,14 @@ impl LedgerRecoveryData {
             .find(|qc| qc.certified_block().id() == root_block.id())
             .ok_or_else(|| format_err!("No QC found for root: {}", root_id))?
             .clone();
+        // TODO: The earlier assertion could fail when executing blocks based on order votes.
+        // A commit certificate may not have a QC with ledger info that matches with the
+        // commit certificate.
         let root_ordered_cert = quorum_certs
             .iter()
             .find(|qc| qc.commit_info().id() == root_block.id())
+            .map(|qc| qc.into_wrapped_ledger_info())
+            .or_else(|| highest_ordered_cert.filter(|cert| cert.commit_info().id() == root_id))
```

Comment:
I thought about this more, i don't think storing a single highest order cert works, this could be any cert that between highest committed to highest ordered. we need to store all the pending committed ordered cert to guarantee we can find the correct one